src/psyclone/transformations.py

```diff
@@ -2182,17 +2207,17 @@ def apply(self, outer):  # pylint: disable=arguments-differ
         # Move inner to parent:
         inner.parent = parent
         parent.children.insert(index, inner)
-        outer.children.remove(inner)
+        outer.loop_body.children.remove(inner)
 
-        # Move inner's children to outer:
-        for child in inner.children:
-            inner.children.remove(child)
-            outer.children.append(child)
-            child.parent = outer
+        # Move inner's schedule to outer:
+        outer.children[3] = inner.loop_body
+        inner.children[3] = Schedule()
```

Comment:
Do you have to create this empty Schedule for the Loop internal-consistency checks? Please add a comment.

Reply:
It is simply to remove all current entries (an alternative to loop through the body and `del` each children). I have reorganized and commented the code.